src/components/app/ListOfPublishedProfiles.js

```diff
@@ -283,19 +283,33 @@ type State = {|
 |};
 
 export class ListOfPublishedProfiles extends PureComponent<Props, State> {
+  _isMounted = false;
+
   state = {
     profileDataList: null,
   };
 
   async componentDidMount() {
+    this._isMounted = true;
     const profileDataList = await listAllProfileData();
+    window.addEventListener('focus', this.loadAllProfileData);
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    window.removeEventListener('focus', this.loadAllProfileData);
+  }
 
+  loadAllProfileData = async () => {
+    const profileDataList = await listAllProfileData();
     // It isn't ideal to use a setState here, but this is the only way.
     // eslint-disable-next-line react/no-did-mount-set-state
```

Comment:
probably you'll need to move this comment inside the `if` block so that the eslint rule is disabled at the right place.